app/src/main/java/com/davidtomas/taskyapp/features/agenda/data/agenda/repository/AgendaRepositoryImpl.kt

```diff
@@ -41,30 +44,86 @@ class AgendaRepositoryImpl(
         return coroutineScope {
             agendaService.getAgenda().map {
                 val job1 = this.launch(Dispatchers.IO) {
-                    eventLocalSource.saveEvents(
+                    val job11 = launch {
+                        eventLocalSource.saveEvents(
+                            it.eventResponses.map {
+                                it.toEventModel()
+                            }
+                        )
+                    }
+                    val job12 = launch {
                         it.eventResponses.map {
                             it.toEventModel()
+                        }.filter { it.remindAt > System.currentTimeMillis() }.forEach {
+                            notificationScheduler.cancelScheduledNotificationAndPendingIntent(it)
+                            notificationScheduler.scheduleNotification(it)
                         }
-                    )
+                    }
+                    job11.join()
+                    job12.join()
                 }
```

Comment:
What is the point of this outer job?
Also, naming the jobs based on what they do could be more readable, such as `saveEventJob`